server/src/main/java/org/folio/reservoir/server/OaiPmhClientService.java

```diff
@@ -520,22 +515,30 @@ Future<Boolean> ingestRecord(
     }
   }
 
-  Future<HttpClientResponse> listRecordsRequest(JsonObject config) {
-    RequestOptions requestOptions = new RequestOptions();
-    requestOptions.setMethod(HttpMethod.GET);
-    requestOptions.setHeaders(getHttpHeaders(config));
-    QueryStringEncoder enc = new QueryStringEncoder(config.getString("url"));
-    enc.addParam("verb", "ListRecords");
-    if (!addQueryParameterFromConfig(enc, config, RESUMPTION_TOKEN_LITERAL)) {
-      addQueryParameterFromConfig(enc, config, "from");
-      addQueryParameterFromConfig(enc, config, "until");
-      addQueryParameterFromConfig(enc, config, "set");
-      addQueryParameterFromConfig(enc, config, "metadataPrefix");
+  Future<OaiResponse<JsonObject>> listRecordsRequest(OaiPmhStatus job) {
+    JsonObject config = job.getConfig();
+    XmlMetadataStreamParser<JsonObject> metadataParser = new XmlMetadataParserMarcInJson();
+    OaiRequest<JsonObject> oaiRequest = new OaiHttpRequest<>(httpClient, config.getString("url"),
+        metadataParser, config.getBoolean("xmlFixing", false), getHttpHeaders(config));
+    String resumptionToken = config.getString(RESUMPTION_TOKEN_LITERAL);
+    if (resumptionToken != null) {
+      oaiRequest.token(resumptionToken);
+    } else {
+      oaiRequest
```

Comment:
Move to OaiHttpRequest "fromConfig" or "fromJson" method.

Reply:
No. I can make a util for this. The request interface should be minimal. There will be another one for storage

Reply:
Ok, I will wait for that implementation then. I am not sure I get the concept of using those interfaces for SOLR export.

Reply:
Yes, some sort of factory is also OK for this.